controllers/runnerpool_controller.go

```diff
@@ -172,6 +166,48 @@ func mergeMap(m1, m2 map[string]string) map[string]string {
 	return m
 }
 
+func (r *RunnerPoolReconciler) reconcileSecret(ctx context.Context, log logr.Logger, rp *meowsv1alpha1.RunnerPool) error {
+	s := &corev1.Secret{}
+	s.SetNamespace(rp.GetNamespace())
+	s.SetName(rp.GetRunnerSecretName())
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, s, func() error {
+		if expiresInStr, ok := s.Annotations[constants.RunnerSecretExpiresInAnnotationKey]; ok {
+			expiresIn, err := time.Parse(time.RFC3339, expiresInStr)
+			if err != nil {
+				log.Error(err, "failed to parse expires-in annotation")
+				return err
+			}
+			if expiresIn.After(time.Now().Add(5 * time.Minute)) {
+				log.Info("skip update secret", "expires_in", expiresIn)
+				return ctrl.SetControllerReference(rp, s, r.scheme)
+			}
+		}
+		runnerToken, err := r.githubClient.CreateRegistrationToken(ctx, rp.Spec.RepositoryName)
```

Comment:
This process may be called many times.
Isn't that a problem?

Reply:
Changed the implementation of communicating with github in the reconcile loop.
Implemented a new SecretWatcher.
The SecretWatcher periodically monitors the secrets associated with the RunnerPool resource and updates the secrets that are about to expire.